src/gc/proxy.d

```diff
@@ -245,7 +245,7 @@ extern (C)
         return instance.runFinalizers( segment );
     }
 
-    bool gc_inFinalizer() nothrow
+    bool gc_inFinalizer() nothrow @nogc
```

Comment:
Note: not adding `@safe` here, because we can't access the `__gshared` variable `instance` in `@safe` code.

Reply:
Well, then should `@safe` be on the prototype? In essence, if we mark it safe there, then it's de-facto trusted. I'd rather see that be explicit. Or, we can omit safe for the time being (this is, after all, a new feature, so omitting safe shouldn't break any code).

But I find it troubling that this couldn't be made safe. How is memory allocation safe, since it is going to access the instance as well?

Reply:
Do this I think:

```D
return (() @trusted => instance)().inFinalizer();
```

Reply:
Rationale: Technically the access to `__gshared` instance is safe because instance is set only when accessing methods that need a real GC (this is how the proxy system works). And it should be safe to use the old instance in that case, because we aren't going to be in a finalizer for that thread anyway.

Note that I think this depends on the load of the instance pointer being atomic. But I think every other method in here assumes that too.

Reply:
If you want to go this road, I guess it would be better to rename `instance` to `_instance` and a `@safe` `instance` property, so that all proxy function can be made `@safe`

Reply:
Yeah, technically, the instance should be loaded atomically. But if that is a problem, it's already a problem, so we can ignore that for now. I'd say a trusted accessor is probably the best route.

Not sure all proxy functions can be safe, but at least the instance fetching can be made safe.

Reply:
@schveiguy done, can you confirm if this is what you were looking for?

Reply:
Ugh, I see you had to mark `profileStats` trusted. But for the same reason. I think it's reasonable.

Thanks.